Assets/Player/Mana.cs

```diff
@@ -0,0 +1,42 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Mana : MonoBehaviour
+{
+    [SerializeField] float maxMana = 100f;
+    [SerializeField] float minMana = 0f;
+    [SerializeField] float currentMana;
+
+    public float _currentMana { get { return currentMana; } }
```

Comment:
usually the convetion is to have private fields and properties start with _ and public ones without, this here is the reverse